crypto/err/err.c

```diff
@@ -838,19 +733,14 @@ const char *ERR_reason_error_string(unsigned long e)
     return ((p == NULL) ? NULL : p->string);
 }
 
-void ERR_remove_thread_state(const CRYPTO_THREADID *id)
+void ERR_remove_thread_state(void *unused)
```

Comment:
That was to avoid breaking the API, but in retrospect it seems unnecessary (we are already breaking it!) so I changed it to `void ERR_remove_thread_state(void)` and updated all the calls to it (they were already using `NULL` anyway, so the behaviour is the same). I also updated the documentation.